handlers/jiochat/jiochat_test.go

```diff
@@ -0,0 +1,366 @@
+package jiochat
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"github.com/nyaruka/courier/config"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/nyaruka/courier"
+	. "github.com/nyaruka/courier/handlers"
+)
+
+type jiochatAccessMock struct {
+	accessToken string
+	err         error
+}
+
+func (ja *jiochatAccessMock) fetchAccessToken(h *handler, channel courier.Channel) error {
+	return ja.err
+}
+
+func (ja *jiochatAccessMock) getAccessToken(h *handler, channel courier.Channel) string {
+	return ja.accessToken
+}
```

Comment:
I would rather see the mock server satisfy these requests so this is all tested, as is we aren't really testing any of the access token stuff.

Reply:
I'll check more on this, that is the issue I had I was not able to test the header was set correctly and used the interface so I can mock it in the tests, I try checking on this suggestion then